### PR TITLE
Survey: Link and Notification

### PR DESCRIPTION
**Pull request by alsturm**

### CODE REVIEW

**Review by thiloSchlemmer — DISMISSED** [review]
Thx!

File: src/modules/survey/components/Survey.js [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+						<div class='survey__notification'>					
+							<div class='survey__notification-section'>
+								<i class='survey__notification-icon'></i>
+								<div>
+									<div class='survey__notification-primary-text' >${translate('survey_notification_header')}</div>
+									<div class='survey__notification-secondary-text' >${translate('survey_notification_text')}</div>
+								</div>
+							</div>
+							<div class='survey__notification-section space-evenly'>							
+								<ba-button id='closeButton' .label=${'Nein Danke'} @click=${onClose}></ba-button>
```
Comment (taulinger) [inline src/modules/survey/components/Survey.js:98]: `.label=${'Nein Danke'}` <= no i18n

File: src/modules/survey/components/Survey.js [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+	createView(model) {
+		const { isPortrait, hasMinWidth, isOpen, hasBeenVisible } = model;
+
+
+		const getOrientationClass = () => {
+			return isPortrait ? 'is-portrait' : 'is-landscape';
+		};
+
+		const getMinWidthClass = () => {
+			return hasMinWidth ? 'is-desktop' : 'is-tablet';
```
Comment (taulinger) [inline src/modules/survey/components/Survey.js:72]: `'is-desktop'` is not covered by a test

File: src/modules/notifications/components/NotificationItem.js [hunk 3]
```diff
@@ -109,4 +109,8 @@ export class NotificationItem extends MvuElement {
 		return this._onClose;
 	}
 
+	get close() {
+		return this._hide();
+	}
+
```
Comment (taulinger) [inline src/modules/notifications/components/NotificationItem.js:115]: Do we really need this getter anymore?

File: src/modules/survey/components/Survey.js [hunk 4]
```diff
@@ -0,0 +1,132 @@
...
+								<a target='_blank' href='${translate('survey_link')}' @click=${onClose} class="survey__notification-link">${translate('survey_notification_open')}</a>
+							</div>
+						</div>`;
+			};
+			emitNotification(getContent(), LevelTypes.CUSTOM);
+		};
+		if (!hasBeenVisible) {
+			window.setTimeout(() => showNotification(), Survey_Notification_Delay_Time);
+			this.signal(Update_HasBeenVisible, true);
+		}
```
Comment (taulinger) [inline src/modules/survey/components/Survey.js:108]: Does it make a difference if we just call `window.setTimeout(() => showNotification(), Survey_Notification_Delay_Time);` ?
The model property `hasBeenVisible` is not read anywhere else, we can omit it in my eyes.

File: src/modules/survey/i18n/survey.provider.js [hunk 5]
```diff
@@ -0,0 +1,29 @@
+export const surveyProvide = (lang) => {
+	switch (lang) {
+
+		case 'en':
+			return {
+				//the first part of the snake_case key should be the name of the related module
+				survey_button: 'Survey',
+				survey_notification_header: 'Survey',
+				survey_notification_text: 'What functions would you like to see in the new BayernAtlas',
+				survey_notification_close: 'no Thanks',
```
Comment (taulinger) [inline src/modules/survey/i18n/survey.provider.js:10]: => 'No thanks'

File: test/modules/survey/components/Survey.test.js [hunk 6]
```diff
@@ -0,0 +1,153 @@
...
+
+			const element = await setup(state, {});
+			expect(store.getState().notifications.latest).toBeNull();
+			jasmine.clock().tick(3100);
+
+			expect(element).toBeTruthy();
+			expect(isTemplateResult(store.getState().notifications.latest.payload.content)).toBeTrue();
+		});
+
+
```
Comment (taulinger) [inline test/modules/survey/components/Survey.test.js:105]: Unnecessary new line

File: test/modules/survey/i18n/survey.provider.test.js [hunk 7]
```diff
@@ -0,0 +1,50 @@
...
+		const map = surveyProvide('de');
+
+		expect(map.survey_button).toBe('Umfrage');
+		expect(map.survey_notification_header).toBe('Umfrage');
+		expect(map.survey_notification_text).toBe('Welche Funktionen wünschen Sie sich für den neuen BayernAtlas?');
+		expect(map.survey_notification_close).toBe('Nein Danke');
+		expect(map.survey_notification_open).toBe('Mitmachen');
+		expect(map.survey_link).toBe('https://github.com/ldbv-by/bav4-nomigration');
+	});
+
```
Comment (taulinger) [inline test/modules/survey/i18n/survey.provider.test.js:30]: Unnecessary new line

File: test/modules/survey/i18n/survey.provider.test.js [hunk 8]
```diff
@@ -0,0 +1,50 @@
...
+		const map = surveyProvide('en');
+
+		expect(map.survey_button).toBe('Survey');
+		expect(map.survey_notification_header).toBe('Survey');
+		expect(map.survey_notification_text).toBe('What functions would you like to see in the new BayernAtlas');
+		expect(map.survey_notification_close).toBe('no Thanks');
+		expect(map.survey_notification_open).toBe('Sure');
+		expect(map.survey_link).toBe('https://github.com/ldbv-by/bav4-nomigration');
+	});
+
```
Comment (taulinger) [inline test/modules/survey/i18n/survey.provider.test.js:17]: Unnecessary new line

**Review by taulinger — CHANGES_REQUESTED** [review]
Nice progress!
some minor parts should be improved.
Also, remove the empty file `src/modules/survey/components/index.js` please.

File: test/modules/survey/components/Survey.test.js [hunk 9]
```diff
@@ -0,0 +1,153 @@
...
+		it('emits a notification with a close-Button and a open-Button', async () => {
+			const target = document.createElement('div');
+			const state = {
+				media: {
+					portrait: true
+				}
+			};
+
+			const element = await setup(state, {});
+			jasmine.clock().tick(3100);
```
Comment (taulinger) [inline test/modules/survey/components/Survey.test.js:115]: We should export the delay constant in `Survey.js`, import and use it here: 
`jasmine.clock().tick(Survey_Notification_Delay_Time+100);`

In Survey.js we can also test the value of the constant then.

File: test/modules/survey/components/Survey.test.js [hunk 10]
```diff
@@ -0,0 +1,163 @@
...
+	describe('when initialized', () => {
+		beforeEach(async () => {
+			jasmine.clock().install();
+		});
+
+		afterEach(function () {
+			jasmine.clock().uninstall();
+		});
+
+		it('adds survey for landscape move', async () => {
```
Comment (taulinger) [inline test/modules/survey/components/Survey.test.js:51]: `move` <= 😂

File: src/modules/survey/i18n/survey.provider.js [hunk 11]
```diff
@@ -0,0 +1,29 @@
+export const surveyProvide = (lang) => {
+	switch (lang) {
+
+		case 'en':
+			return {
+				//the first part of the snake_case key should be the name of the related module
+				survey_button: 'Survey',
+				survey_notification_header: 'Survey',
+				survey_notification_text: 'What functions would you like to see in the new BayernAtlas',
+				survey_notification_close: 'no thanks',
```
Comment (taulinger) [inline src/modules/survey/i18n/survey.provider.js:10]: '[N]o thanks'

File: test/modules/survey/components/Survey.test.js (hunk 10 above)
Comment (taulinger) [inline test/modules/survey/components/Survey.test.js:51]: more precise would be something like 'renders  for landscape mode'

let's combine it with `describe('responsive layout '` in line 141

**Review by taulinger — COMMENTED** [review]
Almost done 😀

**Review by taulinger — APPROVED** [review]
Let's go!

**Pull request merged**